src/main/scala/SIFPAstExtensions.scala

```diff
@@ -0,0 +1,101 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.silver.sif
+
+import viper.silver.ast.{ExtensionExp, IntLit, NoPosition, Position}
+import viper.silver.parser.{NameAnalyser, PExp, PExtender, PIntLit, PNode, PType, PTypeSubstitution, Translator, TypeChecker, TypeHelper}
+
+case class PLowExp(e: PExp)(val pos: (Position, Position) = (NoPosition, NoPosition)) extends PExtender with PExp {
+  typ = TypeHelper.Bool
+
+  override def typeSubstitutions = e.typeSubstitutions
+
+  override def forceSubstitution(ts: PTypeSubstitution): Unit = {
+    e.forceSubstitution(ts)
+  }
+
+  override val getSubnodes: Seq[PNode] = Seq(e)
+
+  override def typecheck(t: TypeChecker, n: NameAnalyser, expected: PType): Option[Seq[String]] = {
+    t.checkTopTyped(e, None)
+    if (expected == TypeHelper.Bool)
+      None
+    else
+      Some(Seq(s"Expected type ${expected}, but got Bool"))
```

Comment:
Just making sure that all error messages intentionally do not end with a period.